astropy/coordinates/matrix_utilities.py

```diff
@@ -159,14 +164,20 @@ def is_O3(matrix):
     """
     # matrix is in O(3) (rotations, proper and improper).
     I = np.identity(matrix.shape[-1])
+    if atol is None:
+        if isinstance(matrix.dtype, np.floating):
+            atol = np.finfo(matrix.dtype).eps * 5
```

Comment:
There's a small mistake here. The dtype can't be checked this way because the ``dtype`` is a class, not an instance. The way to check the dtype is with [issubdtype](https://numpy.org/doc/stable/reference/generated/numpy.issubdtype.html#numpy-issubdtype). This should fix the test issue.

Reply:
Thanks! :)

Reply:
May I ask you why this fixed the test? Also , is this line still useful?
https://github.com/astropy/astropy/blob/6002feae6ed05b3a9f7c414bc33ce99e19e47534/astropy/coordinates/tests/test_matrix_utilities.py#L107
Sorry I'm a bit thick...

Reply:
It's just that `float64` is not an instance of a subclass of `floating`.